src/data-sources/connections/application-insights/QueryTester.tsx

```diff
@@ -49,27 +52,42 @@ export default class QueryTester extends React.Component<IQueryTesterProps, IQue
     this.setState({ showDialog: true });
   }
 
+  collapseResponse() {
+    this.setState({ expandResponse: false });
+  }
+
+  expandResponse() {
+    this.setState({ expandResponse: true });
+  }
+
   submitQuery() {
-    this.setState({ loadingData: true, response: '' });
+    this.setState({ loadingData: true, response: {} });
 
     let appInsightsApi = new ApplicationInsightsApi(this.props.applicationID, this.props.apiKey);
-    appInsightsApi.callQuery(this.state.query, json => {
-      // and later turn off indicator
-      var response = JSON.stringify(json);
-      var h = document.getElementById('testerForm').clientHeight - 185;
-      this.setState({ loadingData: false, response: response, responseHeight: h });
+    appInsightsApi.callQuery(this.state.query, (err, json) => {
+      this.setState({ loadingData: false, response: json });
     });
   }
 
   onQueryChange(value: string, event: any) {
     this.setState({ query: value });
   }
 
+  onResponseChange(value: object, event: any) {
+    this.setState({ response: value });
+  }
+
   render() {
-    let { showDialog, query, response, loadingData, responseHeight } = this.state;
+    let { showDialog, query, response, loadingData, expandResponse } = this.state;
 
-    const nav = <Button icon onClick={this.closeDialog}>close</Button>;
-    const action = <Button flat label="Send" onClick={this.submitQuery} />;
+    const dialogActions = [
+            { onClick: this.submitQuery, primary: true, label: 'Run query' },
+            { onClick: this.collapseResponse, primary: false, label: 'Collapse', 
+            disabled: _.isEmpty(response) || !expandResponse ? true : false}, 
```

Comment:
should be:
`_.isEmpty(response) || !expandResponse`
that is the same as `_.isEmpty(response) || !expandResponse ? true : false`

(Here and everywhere - HAE)